### PR TITLE
Nightly alpha builds

### PR DESCRIPTION
**Pull request by graeme**
Task/Issue URL: https://app.asana.com/0/0/1204986042798452/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204986042798431/f
CC: @loremattei @THISISDINOSAUR @ayoy 

**Description**:
Right now, we’re concentrating most of our internal testing on the release candidates. This means that we have a lot fewer eyes on issues that might have been merged much earlier in the cycle. Furthermore, we have to rely on bespoke builds for anything that’s kept only do the Debug build config and rely on people needing to authenticate to see debug options (which means only the latest OS versions).

The ability to build alphas was already merged in #1913 but it still relied on building an distributing locally from Xcode. This PR adds:
- The ability to build Alphas to a specified TestFlight group from GHA
- Nightly Alpha build deployments

**Steps to test this PR**:
1. Take a look at the [most recent successful build](https://github.com/duckduckgo/iOS/actions/runs/6170933936/job/16748187934) from the added action
2. Take a look the DuckDuckGo Alpha TestFlight on App Store Connect and check that the build number from the recent build’s logs is visible there. It’s been distributed to the "Latest Alpha Group” for testing / outside of nightly deployment purposes. Feel free to add yourself to this group and check that you see the build on TestFlight (but that’s just testing TestFlight so probably not necessary).
3. Check that the Alpha build dSyms have been uploaded to the [Asana task](https://app.asana.com/0/0/1205344386326139/f)

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `’`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -94,3 +94,8 @@ jobs:
       with:
         report_paths: unittests.xml
 
+  deploy-alpha:
+    name: Deploy Nightly Alpha Build
+    uses: ./.github/workflows/alpha.yml
```
Comment (ayoy) [inline .github/workflows/nightly.yml:99]: Does this work? :) I think alpha.yml would need a `workflow_call` trigger set up in order to be called from another workflow. Currently it only has `workflow_dispatch` which is not quite the same (this one is used for running workflows manually).

Comment (graeme) [reply .github/workflows/nightly.yml:99]: Ah okay, good to know! Will update that.

Related: wasn’t sure how to test nightly workflows from a branch...

Comment (graeme) [reply .github/workflows/nightly.yml:99]: Fixed! (I think)

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@ name: Nightly Integration Tests
 on:
   schedule:
     - cron: '0 2 * * *' # Run at 2 AM UTC
+  push:
```
Comment (graeme) [inline .github/workflows/nightly.yml]: To be removed before merging

File: .github/workflows/alpha.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    steps:
+
+    - name: Set destination output
+      id: destination
+      run: |
+        INPUT_DESTINATION=${{ github.event.inputs.destination }}
+        echo "destination=${INPUT_DESTINATION}" >> $GITHUB_OUTPUT
+
+    # - name: Assert develop branch TODO: Uncomment before merging
```
Comment (graeme) [inline .github/workflows/alpha.yml]: To be uncommented before merging

**Review by ayoy — APPROVED** [review]
LGTM! Great job @graeme! Remember to remove parts that need to be removed before merging and save your changes 😆

**Pull request merged**